crm_claim_rma/models/crm_claim.py

```diff
@@ -102,60 +99,106 @@ def _onchange_invoice(self):
     @api.onchange('warehouse_id', 'claim_type', 'date')
     def _onchange_invoice_warehouse_type_date(self):
         context = self.env.context
-        claim_line = self.env['claim.line']
         if not self.warehouse_id:
             self.warehouse_id = self._get_default_warehouse()
-        claim_type = self.claim_type
-        claim_date = self.date
-        warehouse = self.warehouse_id
-        company = self.company_id
         create_lines = context.get('create_lines')
 
-        def warranty_values(invoice, product):
-            values = {}
-            try:
-                warranty = claim_line._warranty_limit_values(
-                    invoice, claim_type, product, claim_date)
-            except (InvoiceNoDate, ProductNoSupplier):
-                # we don't mind at this point if the warranty can't be
-                # computed and we don't want to block the user
-                values.update({'guarantee_limit': False, 'warning': False})
-            else:
-                values.update(warranty)
-
-            warranty_address = claim_line._warranty_return_address_values(
-                product, company, warehouse)
-            values.update(warranty_address)
-            return values
-
         if create_lines:  # happens when the invoice is changed
-            claim_lines = []
-            invoices_lines = self.invoice_id.invoice_line_ids.filtered(
-                lambda line: line.product_id.type in ('consu', 'product')
-            )
-            for invoice_line in invoices_lines:
-                location_dest = claim_line.get_destination_location(
-                    invoice_line.product_id, warehouse)
+            self._create_claim_lines()
+
+        if self.invoice_id:
+            self.delivery_address_id = self.invoice_id.partner_id.id
+
+    @api.model
+    def __check_packages_enabled(self):
```

Comment:
api.model decorator makes the method public and callable from xmlrpc which contradicts the double underscore flagging the method as private.